From b89e4c5211e9c11f01ab8985f4ecaa52dc6a0f30 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:50:08 -0400
Subject: [ethernet] bnx2x: Make BP_VF more robust

Message-id: <1409946615-5862-64-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91597
O-Subject: [RHEL7.1 PATCH 63/70] bnx2x: Make BP_VF more robust
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 58fee00ffae3b13b86f801146631012b3db4cea4
Author: Yuval Mintz <Yuval.Mintz@qlogic.com>
Date:   Sun Aug 17 16:47:50 2014 +0300

    bnx2x: Make BP_VF more robust

    Prevent dereference of pointer in case it's NULL.

    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Signed-off-by: Ariel Elior <Ariel.Elior@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
index ca1055f..01bafa4 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
@@ -299,7 +299,8 @@ struct bnx2x_vfdb {
 #define BP_VFDB(bp)  ((bp)->vfdb)
  /* vf array */
  struct bnx2x_virtf *vfs;
-#define BP_VF(bp, idx)  (&((bp)->vfdb->vfs[idx]))
+#define BP_VF(bp, idx)  ((BP_VFDB(bp) && (bp)->vfdb->vfs) ? \
+     &((bp)->vfdb->vfs[idx]) : NULL)
 #define bnx2x_vf(bp, idx, var) ((bp)->vfdb->vfs[idx].var)
 
  /* queue array - for all vfs */
-- 
1.7.1